client/backoff/backoff.go

```diff
@@ -22,37 +22,55 @@ import (
 	"time"
 )
 
-// Backoff specifies the parameters of the backoff algorithm.
+// Backoff specifies the parameters of the backoff algorithm. Works correctly
+// if 0 < Min <= Max <= 2^62 (nanosec), and Factor >= 1.
 type Backoff struct {
-	Min    time.Duration
-	Max    time.Duration
-	Factor float64
-	Jitter bool
-	x      int
+	Min    time.Duration // Duration of the first pause.
+	Max    time.Duration // Max duration of a pause.
+	Factor float64       // The factor of duration increase between iterations.
+	Jitter bool          // Add random noise to pauses.
+
+	delta time.Duration // Current pause duration relative to Min, no jitter.
 }
 
-// Duration returns the time to wait on duration x.
-// Every time Duration is called, the returned value will exponentially increase by Factor
-// until Backoff.Max. If Jitter is enabled, will wait an additional random value between
-// 0 and factor^x * min, capped by Backoff.Max.
+// Duration returns the time to wait on current retry iteration.
+// Every time Duration is called, the returned value will exponentially
+// increase by Factor until Backoff.Max. If Jitter is enabled, will wait an
+// additional random value between 0 and Factor^x * Min, capped by Backoff.Max.
 func (b *Backoff) Duration() time.Duration {
-	// min( min * factor ^ x , max)
-	minNanos := float64(b.Min.Nanoseconds())
-	maxNanos := float64(b.Max.Nanoseconds())
-	nanos := minNanos * math.Pow(b.Factor, float64(b.x))
-	nanos = math.Min(nanos, maxNanos)
+	pause := b.Min + b.delta
+
+	newPause := time.Duration(float64(pause) * b.Factor)
```

Comment:
Add a comment that this is computing the *next* pause, not the current one.

Reply:
Perhaps try to create some visual separation in this function between the current pause and the next one